_posts/en/newsletters/2021-11-10-newsletter.md

```diff
@@ -0,0 +1,225 @@
+  taproot transaction.  The problem has been [patched][neutrino #234]
+  in the Neutrino source code and other implementations of compact block
+  filters are not affected, but Olaoluwa Osuntokun did start a thread on
+  the [Bitcoin-Dev][bd cbf thread] and [LND-Dev][ld cbf thread] mailing
+  lists about the problem and possible future improvements to compact
+  block filters, including:
+
+    - *New filters:* creating additional optional filter types to allow
+      lightweight clients to search for other types of data.
+
+    - *New P2P protocol message:* adding a new P2P protocol message for
+      retrieving block undo data.  Block undo data includes the previous
+      outputs (and related information<!--like heights-->) for each
+      input spent in a block, which can be combined with a block to
+      fully verify a filter was generated from that data.  Undo data can
+      itself be [verified][harding undo verification] in the case of
+      discrepancies between peers.
+
+    - *Multi-block filters:* these could further reduce the data
+      lightweight clients will need to download.
+
+    - *Committed block filters:* requiring miners commit to the filter
+      for their blocks, reducing the amount of data lightweight clients
+      need to download to monitor for discrepancies between the filters
+      being served by different peers.
+
+## Bitcoin Core PR Review Club
+
+*In this monthly section, we summarize a recent [Bitcoin Core PR Review Club][]
+meeting, highlighting some of the important questions and answers.  Click on a
+question below to see a summary of the answer from the meeting.*
+
+[Add `ChainstateManager::ProcessTransaction`][review club #23173] is a PR by
+John Newbery to add a new `ChainstateManager::ProcessTransaction()` interface
+function responsible for processing transactions as candidates to the mempool
+and performing mempool consistency checks. The review club discussed the
+current interface for adding transactions to the mempool.
+
+{% include functions/details-list.md
+  q0="What is `cs_main`? Why is it called `cs_main`?"
+  a0="`cs_main` is a mutex intended to synchronize multi-threaded access to
+validation state. In reality, it also guards non-validation data in
+net\_processing, and multiple contributors wish to minimize the usage of`cs_main`.
+The variable was named when validation functionality was housed in a main.cpp
+file. The prefix, `cs`, stands for critical section."
+a0link="https://bitcoincore.reviews/23173#l-45"
+
+  q1="Which components currently call `AcceptToMemoryPool`? Which of the ATMP calls are from external client code and which are from inside validation?"
+  a1="Excluding calls from tests, there are four call sites:
+  1. When the node starts, it [loads][atmp disk] transactions from mempool.dat and
+     calls ATMP to re-validate the transactions and restore mempool contents. This
+     call originates from a component 'external' to validation.
```

Comment:
I disagree that this is external to validation. `LoadMempool()` is contained within validation.cpp and is called from a `CChainState` object (see https://bitcoincore.reviews/23173#l-107).

Reply:
I think this was a typo - thanks!